packages/app/src/client/services/Picker/PickerService.tsx

```diff
@@ -154,6 +155,10 @@ function PickerService({ children }: { children: React.ReactNode }) {
       </pickerContext.Provider>
       <Dialog
         data-testid="dialog"
+        data-test={
+          "pickerService/dialog" +
+          (activePicker?.dataTest ? "/" + activePicker.dataTest : "")
+        }
```

Comment:
In my experience relying too much on data attributes in E2E tests can cause problems later on. Have you considered to rather rely on semantics to target the DOM? E.g. role/labels/text?
 
Not sure if playwright makes that easy, but it would be the equivalent of testing-library's `getByRole` et al.

Reply:
Just found something about this in the [playwright's best practices](https://playwright.dev/docs/selectors/#prioritize-user-facing-attributes).

Reply:
As a general rule of thumb, if a button moves slightly, or content/text changes these browser tests shouldn't fail or need to be changed. Testing against labels/text etc... can be a huge maintenance nightmare during refactors. I've found putting explicit data attributes in for the testing to use to be the way to go - as described in your link as "Define explicit contract".